xonsh/xoreutils/_which.py

```diff
@@ -191,21 +192,21 @@ def whichgen(command, path=None, verbose=0, exts=None):
     # Windows has the concept of a list of extensions (PATHEXT env var).
     if sys.platform.startswith("win"):
         if exts is None:
-            exts = os.environ.get("PATHEXT", "").split(os.pathsep)
+            exts = builtins.__xonsh_env__['PATHEXT']
```

Comment:
> Note that we have made minor modifications to code that we have forked from IPython, such as inpectors, to be able to function with xonsh correctly.

We can try to extract these in separate packages. I know Xonsh vendors all necessary things, but IPython have lot's of dependencies, so I have no real issues with make separate pacakges.